lib/dotcom/trip_plan/itinerary_group.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Dotcom.TripPlan.ItineraryGroup do
+  @moduledoc """
+  A single group of related itineraries
+  """
+
+  alias Dotcom.TripPlan.Itinerary
+
+  defstruct [:itineraries, :representative_index, :representative_time, :summary]
+
+  @type summarized_leg :: %{
+          routes: [Routes.Route.t()],
+          walk_minutes: non_neg_integer()
+        }
+
+  @type summary :: %{
+          accessible?: boolean() | nil,
+          duration: non_neg_integer(),
+          start: DateTime.t(),
+          stop: DateTime.t(),
+          summarized_legs: [summarized_leg()],
+          tag: String.t(),
+          total_cost: non_neg_integer(),
+          walk_distance: float()
+        }
+
+  @type t :: %__MODULE__{
+          itineraries: [Itinerary.t()],
+          representative_index: non_neg_integer(),
+          representative_time: :start | :stop,
```

Comment:
Nit (non-blocking): The name `representative_time` makes me think that the value stored here will be a time, but it's actually an atom indicating where to look for an itinerary's time. Maybe `representative_time_key`?